### ISSUE TITLE
Can't run test cases against installed meson due to changes in how meson is detected in mesonlib.py

### ISSUE THREAD

**Issue by DimStar77** [mesonbuild/meson#3015]
I'm backaging meson for a distro, and naturally, I'm interested to monitor that meson, the package I provide, works.

Up to version 0.43.0, I carried a small patch on the test suite that forced the usage of /usr/bin/meson instead of the local sources (the directories mesonbuild and the file meson.py were removed before running the test suite, to ensure the distro provided files are used)

With 0.44.0, this is no longer possible - as mesonbuild/mesonnlib.py has code to 'detect how meson is being started' - and fails in pretty much any way possible.

The simplest reproducer is ```python3 -c 'from mesonbuild import mesonlib'``` which results in an error:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.6/site-packages/mesonbuild/mesonlib.py", line 63, in <module>
    meson_command = python_command + [detect_meson_py_location()]
  File "/usr/lib/python3.6/site-packages/mesonbuild/mesonlib.py", line 54, in detect_meson_py_location
    raise RuntimeError('Could not determine how to run Meson. Please file a bug with details.')
RuntimeError: Could not determine how to run Meson. Please file a bug with details.
```

It would be really great if the test suite could be run with the installed meson as test target; this allows distro packagers to validate their package being correct.

**Comment by nirbheek** [comment]
Your fix was always wrong; even if `/usr/bin/meson` was being used, the unit tests would still use the meson modules inside the source tree.

I suppose the correct way to fix this would be to add an option to `run_tests.py` to search for meson on the system (in `PATH` and via the system python module path).

**Comment by nirbheek** [comment]
fwiw, I can't reproduce the error you listed with any version of Meson. How are you running it?

**Comment by DimStar77** [comment]
As stated in my inital message: meson.py and the directory mesonbuild were removed before runnjng the testsuite, to ensure the installed files are being used

Simpy start the python command mentioned in the initial comment - anywhere outside a git checkout of meson. It is impossible to import meson in anything, unless there is a meson.py in cwd

**Comment by nirbheek** [comment]
I see now, this will be fixed for 0.44.1, and in 0.45
